jetty-util/src/main/java/org/eclipse/jetty/util/component/Container.java

```diff
@@ -114,12 +134,35 @@
      */
     boolean isManaged(Object bean);
 
+    /**
+     * @param clazz the class of the beans
+     * @param <T> the Bean type
+     * @return the list of beans of the given class from the entire Container hierarchy
+     */
+    <T> Collection<T> getContainedBeans(Class<T> clazz);
+
+    /**
+     * Get the beans added to the connector that are EventListeners.
```

Comment:
What's a connector got to do with the Container class?

Reply:
oops